tools/api-markdown-documenter/api-report/api-markdown-documenter.api.md

```diff
@@ -292,7 +288,24 @@ export interface DocumentNodeProps {
     readonly frontMatter?: string;
 }
 
-export { DocumentWriter }
+// @public
+export interface DocumentWriter {
```

Comment:
Would we rather have this be `IDocumentWriter` to stick with the conventions in the rest of the repo? And what's the motivation to move from free function to namespace?

Reply:
Since I'm also using it as a namespace, using interface naming seemed odd (calling `IDocumentWriter.foo` feels strange).

The main motivation for the change is discoverability - if you know how to find `DocumentWriter`, then you can easily find how to create one (intellisense, etc.), rather than having to search for some function by name. API docs can point to the function either way, but the namespace approach has better alternative discoverability options.

This was a somewhat experimental approach, so feedback is appreciated.

Reply:
And note that the use of the `I` prefix (per our guidelines) is only intended for types that are intended to be implemented externally, which is not the case here.

https://github.com/microsoft/FluidFramework/wiki/Coding-Guidelines